### DISCUSSION TITLE
Linux - Add -DNO_LCHMOD macro option to unzip.rb

### DISCUSSION THREAD

**Discussion by mklinkj** [Homebrew/discussions#4445] [Linux]
### Additional Info

- [X] I am using Windows Subsystem for Linux

### Output of `brew config`

```shell
N/A
```

### Output of `brew doctor`

```shell
N/A
```

### Description of issue

Hello.

I am using `brew` in Windows WSL2 Ubuntu environment and I also installed `unzip` using `brew`.

In such an environment, installing Java 17 Temurin using [SDKMan](https://sdkman.io/) always gives the error below.

```
...
lchmod (file attributes) error: Operation not supported
lchmod (file attributes) error: Operation not supported
lchmod (file attributes) error: Operation not supported
...
```

It was said that lchmod could not be used in the Ubuntu environment anyway, so I modified the `brew` file in the local environment.

* /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/unzip.rb

  ```ruby
    def install
      # These macros also follow Ubuntu, and are required:
      # - to correctly handle large archives (> 4GB)
      # - extract & print archive contents with non-latin characters
      loc_macros = %w[
        -DLARGE_FILE_SUPPORT
        -DUNICODE_SUPPORT
        -DUNICODE_WCHAR
        -DUTF8_MAYBE_NATIVE
        -DNO_WORKING_ISPRINT
        -DNO_LCHMOD     
      ]
  ```

  * I added the `-DNO_LCHMOD` macro at the end.
  * https://github.com/Homebrew/homebrew-core/blob/master/Formula/unzip.rb

After adding the macro as above, I deleted the `unzip` and reinstalled it with the modified `unzip.rb`.

```sh
brew uninstall --ignore-dependencies unzip

brew install --build-from-source /home/linuxbrew/.linuxbrew/Homebrew/Library/Taps/homebrew/homebrew-core/Formula/unzip.rb
```

Installing Java 17 Temurin using SDKMan no longer caused any errors.

Is it okay to use it like this?

I thought it would be nice to conditionally include the `-DNO_LCHMOD` macro on Linux systems that don't support `lchmod`... I'm not sure if that's a good idea.

Thanks.

**Comment by cstamas** [comment]
Same problem exists even today (original sdkman issue: https://github.com/sdkman/sdkman-cli/issues/790)
